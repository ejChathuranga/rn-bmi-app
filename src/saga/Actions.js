const ADD_TEST_VALUE = 'ADD_TEST_VALUE'
const JOB_DATA_REQUEST = 'JOB_DATA_REQUEST'
const JOB_DATA_RECEIVED = 'JOB_DATA_RECEIVED'

const addTestValue = action => ({
    type: ADD_TEST_VALUE,
    payload: action
})

const jobsRequestSend = () =>({
    type: JOB_DATA_REQUEST
})

const jobDataReceived = jobs =>({
    type: JOB_DATA_RECEIVED,
    payload: jobs
})


export{
    ADD_TEST_VALUE,
    JOB_DATA_REQUEST,
    JOB_DATA_RECEIVED,
    addTestValue,
    jobsRequestSend,
    jobDataReceived
}