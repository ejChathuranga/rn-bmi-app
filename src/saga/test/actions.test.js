/* describe, it, expect */
import * as actions from '../Actions'

describe('jobDataReceived return actions' , ()=>{
    it('handles a object', ()=>{
        expect(actions.jobDataReceived({'name': 'EJ Chathuranga'})).toMatchSnapshot()
    })
    it('handles a empty object', ()=>{
        expect(actions.jobDataReceived({})).toMatchSnapshot()
    })
    it('handles a object contains empty value', ()=>{
        expect(actions.jobDataReceived({'name': ''})).toMatchSnapshot()
    })
    it('handles a object contains empty value and key', ()=>{
        expect(actions.jobDataReceived({'': ''})).toMatchSnapshot()
    })
})
