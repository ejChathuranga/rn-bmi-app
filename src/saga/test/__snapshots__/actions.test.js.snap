// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jobDataReceived return actions handles a empty object 1`] = `
Object {
  "payload": Object {},
  "type": "JOB_DATA_RECEIVED",
}
`;

exports[`jobDataReceived return actions handles a object 1`] = `
Object {
  "payload": Object {
    "name": "EJ Chathuranga",
  },
  "type": "JOB_DATA_RECEIVED",
}
`;

exports[`jobDataReceived return actions handles a object contains empty value 1`] = `
Object {
  "payload": Object {
    "name": "",
  },
  "type": "JOB_DATA_RECEIVED",
}
`;

exports[`jobDataReceived return actions handles a object contains empty value and key 1`] = `
Object {
  "payload": Object {
    "": "",
  },
  "type": "JOB_DATA_RECEIVED",
}
`;
